realm/realm-library/src/main/java/io/realm/ObservableCollection.java

```diff
@@ -0,0 +1,10 @@
+package io.realm;
+
+/**
+ * Interface for collections that are able to tell interested parties how they
+ * changed when they where updated.
+ */
+public interface ObservableCollection<T, L> extends Observable<T> {
+    void addChangeListener(L listener);
```

Comment:
I'm future-proofing this for other collection types, e.g. Set and Map are not ordered so if they want to expose their own change notification we don't need to have a `ObservableOrderedCollection` and ObservableUnorderdCollection` interface.